packages/opentelemetry-exporter-stackdriver-trace/src/trace.ts

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExportResult } from '@opentelemetry/base';
+import { NoopLogger } from '@opentelemetry/core';
+import { ReadableSpan, SpanExporter } from '@opentelemetry/tracing';
+import { Logger } from '@opentelemetry/types';
+import { GoogleAuth } from 'google-auth-library';
+import { google } from 'googleapis';
+import { StackdriverExporterOptions } from './external-types';
+import { getReadableSpanTransformer } from './transform';
+import { Span, SpansWithCredentials } from './types';
+
+google.options({ headers: { 'x-opentelemetry-outgoing-request': 1 } });
+
+/**
+ * Format and sends span information to StackDriver Trace.
+ */
+export class StackdriverTraceExporter implements SpanExporter {
+  private _projectId?: string;
```

Comment:
We can make it readonly. It is optional because the user may misconfigure and not put it in the options, so in that case we don't throw we just fail to export keeping with the "never throw" rule established for plugins.

Reply:
But without `projectId` it is not possible to export to the backend, which means it should be provided by the user (as a mandatory param). In case of wrong id, we are already logging the error ( and not throwing exception).